### ISSUE TITLE
Request for fine grain image requeriments..

### ISSUE THREAD

**Issue by oscarbg** [google/clspv#19]
You say
"If images are used in the OpenCL C:
    The shaderStorageImageReadWithoutFormat field of VkPhysicalDeviceFeatures must be set to true.
    The shaderStorageImageWriteWithoutFormat field of VkPhysicalDeviceFeatures must be set to true.
"
also seeing vulkan.gpuinfo reports situation is better than int16 support at least modern NV and AMD GPUs support both reqs.. 
but Intel iGPUs don't support shaderStorageImageReadWithoutFormat neither Windows nor Linux driver..

So as OpenCL has __read_only and __write_only image access qualifiers maybe we can require only shaderStorageImageReadWithoutFormat in __read_only images and shaderStorageImageWriteWithoutFormat in __write_only images .. is that correct?

that way a kernel having only write only image access should work on Intel GPUs right?

thanks..

**Comment by oscarbg** [comment]
may have been to fast..
seeing test files included seems includes only what needed..
feel free to close if that's true..

**Comment by sheredom** [comment]
@oscarbg yup! We'll only define the capability when we require it - so a kernel with write_only images would work on Intel GPUs!

**Comment by dneto0** [comment]
See also https://github.com/google/clspv/issues/5 which documents how the compiler is being overly conservative about declaring those capabilities.

**Comment by rjodinchr** [comment]
Gemini:
> clspv only declares storage image capabilities when required by `__read_only` or `__write_only` qualifiers, which resolves this request.

Please reopen if this is still relevant
